bazel/BUILD.bazel

```diff
@@ -0,0 +1,33 @@
+## Copyright 2020 Google LLC
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     https://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+load("//bazel:defs.bzl", "grpc_web_toolchain")
+
+# The toolchain type used to distinguish proto toolchains.
+toolchain_type(
+    name = "toolchain_type",
+    visibility = ["//visibility:public"],
+)
+
+grpc_web_toolchain(
+    name = "closure_toolchain_impl",
+    generator = "//javascript/net/grpc/web:protoc-gen-grpc-web",
+    runtime_library = "//javascript/net/grpc/web:closure_grpcweb_runtime",
+)
```

Comment:
I guess my question here is that, do the attribute `generator` and `runtime_library` have special meaning here? Or it can be any string?

Reply:
Both attributes are Labels that have to point to targets (just like, e.g, `cc_library`'s `deps`). `generator` also must be an executable target (e.g. `{cc,go,java}_binary`) and `runtime_library` must point to a `closure_js_library` (compatible) target.